lib/web_ui/lib/src/engine/window.dart

```diff
@@ -99,8 +99,10 @@ class EngineWindow extends ui.Window {
   /// Simulates clicking the browser's back button.
   Future<void> webOnlyBack() => _browserHistory.back();
 
+  String _defaultRouteName;
+
   @override
-  String get defaultRouteName => _browserHistory.currentPath;
+  String get defaultRouteName => _defaultRouteName ??= _browserHistory.currentPath;
```

Comment:
Should it be frozen on construction, or when the browser window opens rather than on first access?

Reply:
We want to delay the initialization of `defaultRouteName` a bit so that the app has time to initialize the engine and [set `locationStrategy`](https://github.com/flutter/engine/blob/b454251f03f27ff422b2c197761493a9e3133909/lib/web_ui/lib/src/ui/initialization.dart#L12) which is used by `_browserHistory` to [get the current path](https://github.com/flutter/engine/blob/b454251f03f27ff422b2c197761493a9e3133909/lib/web_ui/lib/src/engine/history.dart#L56).

But once the app gets the value of the default route name, it shouldn't change after that.